src/data/nat/log.lean

```diff
@@ -64,4 +93,19 @@ end
 lemma pow_log_le_self (b x : ℕ) (hb : 1 < b) (hx : 1 ≤ x) : b ^ log b x ≤ x :=
 by rw [pow_le_iff_le_log _ _ hb hx]
 
+lemma log_le_log_of_le {b n m : ℕ} (h : n ≤ m) : log b n ≤ log b m :=
+begin
+  by_cases hb : b ≤ 1, by rw log_eq_zero_of_lt' hb; exact zero_le',
+  by_cases hn : n = 0, by rw [hn, log_zero_eq_zero]; exact zero_le',
+
+  rw ←pow_le_iff_le_log _ _ (not_le.mp hb) (lt_of_lt_of_le (zero_lt_iff.mpr hn) h),
+  exact (pow_log_le_self b n (not_le.mp hb) (zero_lt_iff.mpr hn)).trans h,
```

Comment:
I'd suggest `rw ←pos_iff_ne_zero at hn` and `rw not_le at hb` to avoid doing that twice. Alternatively, you could replace

```lean
by_cases hn : n = 0,
```
by
```lean
cases eq_zero_or_pos n with hn hn,
```

and probably similar for `hb`.